searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp

```diff
@@ -37,6 +41,40 @@ void tag_match_data(const HandleRecorder::HandleSet &handles, MatchData &match_d
     }
 }
 
+std::unique_ptr<DegradationParams>
+getDegradationParams(const RankSetup & rankSetup, const Properties & rankProperties)
+{
+    vespalib::string limit_attribute = DegradationAttribute::lookup(rankProperties);
+    size_t limit_maxhits = DegradationMaxHits::lookup(rankProperties);
+    if (!limit_attribute.empty() && limit_maxhits > 0) {
+        bool limit_ascending = DegradationAscendingOrder::lookup(rankProperties);
+        double limit_max_filter_coverage = DegradationMaxFilterCoverage::lookup(rankProperties);
+        double samplePercentage = DegradationSamplePercentage::lookup(rankProperties);
+        double postFilterMultiplier = DegradationPostFilterMultiplier::lookup(rankProperties);
+        return std::make_unique<DegradationParams>(limit_attribute, limit_maxhits, !limit_ascending,
+                                                   limit_max_filter_coverage, samplePercentage, postFilterMultiplier);
+    } else if (rankSetup.hasMatchPhaseDegradation()) {
+        return std::make_unique<DegradationParams>(rankSetup.getDegradationAttribute(), rankSetup.getDegradationMaxHits(),
+                                                   !rankSetup.isDegradationOrderAscending(), rankSetup.getDegradationMaxFilterCoverage(),
+                                                   rankSetup.getDegradationSamplePercentage(), rankSetup.getDegradationPostFilterMultiplier());
+    }
+    return std::unique_ptr<DegradationParams>();
+}
+
+DiversityParams
```

Comment:
it feels wrong to not use the same strategy for both paramer types...

Reply:
Fixed